addons/web/static/src/js/core/dialog.js

```diff
@@ -24,6 +24,12 @@ var Dialog = Widget.extend({
     }),
     events: _.extend({} , Widget.prototype.events, {
         'keydown .modal-footer button':'_onFooterButtonKeyDown',
+        'click.bs.dropdown .o_search_options .dropdown-menu': function (e) {
+            // inside a modal, the fix at 4601b044a6e4dbf162cfca9bf0be5ff2a21dd4f5
+            // won't apply, because the modal is not a child of control panel
+            // The logic of the event handler is good though
+            e.stopPropagation();
+        },
```

Comment:
I would check with the framework team (@aab-odoo is never busy) but it seems like this should be generic on the webclient and not repeated everywhere, even less on all dialogs, even less with an inline function. Also `click.bs.dropdown` is confusing as it seems to be a special event from bootstrap but it is not (it is just click with a jquery namespace...). Also the commit https://github.com/odoo/odoo/commit/4601b044a6e4dbf162cfca9bf0be5ff2a21dd4f5 is kinda wrong, we should use the `events` property (convention but this can also have an impact for event binding/unbinding).